### ISSUE TITLE
Unable to write variables to .env file 

### ISSUE THREAD

**Issue by kalidyasin** [rest-nvim/rest.nvim#349]
it  runs :REST run successfully but it is not writing the response variable in .env file

```http
POST {{URL}}/api/signin
Content-Type: application/json

{
    "email": "{{USERNAME}}",
    "password": "{{PASSWORD}}"
}
--{%

local body = context.json_decode(context.result.body)

-- These environment variables are stored in 'vim.env'
context.set_env("TOKEN", body._token)
context.set_env("studentid", body.student_id)
context.set_env("certificate", body.certificate_type)
context.set_env("userid", body.user)

--%}
``` 
.env file
```env
URL=http://127.0.0.1
USERNAME=test@test.com
PASSWORD=password
```
http file
![silicon_20240406_184843](https://github.com/rest-nvim/rest.nvim/assets/49758338/64d8f6ae-9ae9-4050-8653-dadf51492e51)
env file 
![silicon_20240406_185024](https://github.com/rest-nvim/rest.nvim/assets/49758338/ecf03679-d91f-4daa-987a-6057391a69f4)

**Comment by harrylowkey** [comment]
I have the same issue like you, have you fixed it?

**Comment by kalidyasin** [comment]
no it is not fixed

**Comment by filipiz** [comment]
it will not save the variable to the file.
it will set the variable to vim.env so you can use it.

if you run `:lua print vim.env.userid` it should print whatever comes in the response.

**Comment by NTBBloodbath** [comment]
Storing variables in an env file was kinda problematic before due to the implementation of it. I'll give it a new life once I get some vacations from exams in the uni and hopefully that one will be good enough :)

**Comment by kalidyasin** [comment]
> it is described in the test example:
> 
> https://github.com/rest-nvim/rest.nvim/blob/f96edb54a2940322bc7ed81a1031be04db7d3a99/tests/reqs/script_vars/script_vars.http#L9
> 
> it will not save the variable to the file. it will set the variable to vim.env so you can use it.
> 
> if you run `:lua print(vim.env.userid)` after executing the request, it should print whatever comes in the response.

yes but it is not working if the value is too long like token

**Comment by filipiz** [comment]
It's working for me. I'm storing JWT tokens in my cases, they are quite long.
If there is any limitation on `nvim.env` vars it may be detailed on nvim help files, I guess.

**Comment by filipiz** [comment]
I really enjoyed the current solution, @NTBBloodbath.
Although I miss the possibility to run a script before the curl request is executed.

I have a use case where I have to generate a "signature" header, that is a hash based on request params and body.

I have hacked the previous version of the plugin to make something similar work to me.
On that hack, I set a couple headers like this:

`X-Rest-Nvim-Pre-Script: signature.lua`
`X-Rest-Nvim-Post-Script: set_user.lua`

and these scripts were executed before and after the request.

Never had the chance to polish and open a pull request.
And on this version you have already solved half of the problem.

I'll try to make a pull request as soon as I can find some time to make something useful.

Good luck on your exams!
